### PR TITLE
Changelog entry for v0.4.0

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR is a draft changelog entry for v0.4.0.

**Preview**: https://pygmt-git-changelog-release-gmt.vercel.app/changes.html#changelog

Previous release at #1047

Changelog made by following the instructions at https://github.com/GenericMappingTools/pygmt/blob/master/doc/maintenance.md#updating-the-changelog

To do:
- [x] Add a new entry in doc/_static/version_switch.js for documentation switcher
- [x] Update citation information https://github.com/GenericMappingTools/pygmt#citing-pygmt
- [x] Add the documentation link https://github.com/GenericMappingTools/pygmt#documentation-for-other-versions
- [x] Add compatibility information https://github.com/GenericMappingTools/pygmt#compatibility-with-python-and-gmt-versions
- [x] Add highlights
- [x] Update with any remaining PRs merged before release.
- [x] Edit the change list to remove any trivial changes (updates to the README, typo fixes, CI configuration, etc).
- [x] Edit the list of people who contributed to the release, linking to their GitHub account. Sort their names by the number of commits made since the last release (e.g., use git shortlog HEAD...v0.1.2 -sne).

Relates to https://github.com/GenericMappingTools/pygmt/issues/1333

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by meghanrjones** [comment]
For updating the citation information, the AUTHORSHIP.md file says that "The order of authors will be defined by the number of commits to the repository (git shortlog -sne). The order can also be changed on a case-by-case basis." How do I know whether any of the current order reflects a case-by-case basis or whether I should update based on `git shortlog -sne`?

**Comment by weiji14** [comment]
> How do I know whether any of the current order reflects a case-by-case basis or whether I should update based on `git shortlog -sne`?

See previous discussion at https://github.com/GenericMappingTools/pygmt/pull/1047#discussion_r593880470. Will leave it up to you to take a first stab at the order :) Just to clarify, that line refers to the Zenodo citation order. The release note authors order should go strictly by `git shortlog` (and remove the bots).

**Comment by meghanrjones** [comment]
> > How do I know whether any of the current order reflects a case-by-case basis or whether I should update based on `git shortlog -sne`?
> 
> See previous discussion at [#1047 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1047#discussion_r593880470). Will leave it up to you to take a first stab at the order :) Just to clarify, that line refers to the Zenodo citation order. The release note authors order should go strictly by `git shortlog` (and remove the bots).

OK, I guess I wasn't paying good attention for that release :) Thanks for the note.

**Comment by meghanrjones** [comment]
The index.html result from the `python setup.py --long-description | rst2html.py --no-raw > index.html` step does not render the badges correctly. Does this need to be corrected?

image

**Comment by meghanrjones** [comment]
I did the contributors section based on the git log, but I would like to add the three others who were involved in https://github.com/GenericMappingTools/pygmt/pull/1070 and https://github.com/GenericMappingTools/pygmt/pull/1193 since this method does not recognize team programming. I will do this early tomorrow if there are no objections.

**Comment by seisman** [comment]
> The index.html result from the `python setup.py --long-description | rst2html.py --no-raw > index.html` step does not render the badges correctly. Does this need to be corrected?
> 
> image

I don't think it's an issue. The badges are working on the pypi site (https://pypi.org/project/pygmt/) and we don't make changes to badges since v0.3.1.

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:119]: > I did the contributors section based on the git log, but I would like to add the three others who were involved in #1070 and #1193 since this method does not recognize team programming. I will do this early tomorrow if there are no objections.

Yes! Please add them in, we need to find a way to count co-authors too. I had a brief search at stackoverflow but there doesn't seem to be a good solution somehow. Closest thing I found was https://stackoverflow.com/questions/61263083/how-to-query-git-log-for-co-authored-by/61264586#61264586 but using `git log --format="%(trailers:key=Co-authored-by)"` doesn't give the counts :sweat_smile:

```suggestion
* [Claire Klima](https://github.com/cklima616)
* [Megan Munzek](https://github.com/munzekm)
* [Michael Neumann](https://github.com/MichaeINeumann)
* [Nathan Loria](https://github.com/Nathandloria)
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:108]: There's a couple of maintenance PRs we can probably leave off the changelog, e.g. PRs 1324, 1291?

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:11]: Add the "Tab Autocompletion" to Highlights?

File: README.rst
Comment (weiji14) [inline README.rst:239]: Is the intention of changing 0.4.x to 0.4.0 here due to https://github.com/GenericMappingTools/pygmt/issues/1333#issuecomment-863524502? I.e. we want to pin PyGMT v0.4.1 to GMT 6.2.1?

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:27]: Sort the entries so that the "Wrap ..." are put together.

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:55]: Sort the entries so that the "Deprecate parameter ..." are put together.

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:61]:
```suggestion
```

File: README.rst
Comment (meghanrjones) [reply README.rst:239]: No, just was my interpretation of the checklist.

File: doc/changes.md
Comment (meghanrjones) [reply doc/changes.md]: Sorting all done in https://github.com/GenericMappingTools/pygmt/pull/1341/commits/430e7d96992d757c18234c077903611e7e0c8562

File: doc/changes.md
Comment (meghanrjones) [reply doc/changes.md:12]: Added in https://github.com/GenericMappingTools/pygmt/pull/1341/commits/d06e0d469ff2e06f6be129c13478d51bbe4b45e6

File: doc/changes.md
Comment (meghanrjones) [reply doc/changes.md]: OK, I removed those. The others seem OK to me, but feel free to post if there are others that you see as extraneous.

File: README.rst
Comment (weiji14) [reply README.rst:239]: Ah I see. I think we can revert this one line change as the PyGMT v0.4.x series should stick with these GMT/Python/NumPy versions (patch versions shouldn't update minimum Python/NumPy dependencies per [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy.html#detailed-description)).

```suggestion
0.4.x   >=6.2.0    >=3.7     >=1.17.0
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:12]:
```suggestion
* Minimum required GMT version is now 6.2.0 or newer ([#1321](https://github.com/GenericMappingTools/pygmt/pull/1321))
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:97]: Once something is added as a highlight item, it should be removed from the list below.

```suggestion
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:11]:
```suggestion
* Support tab auto-completion in Jupyter ([#1282](https://github.com/GenericMappingTools/pygmt/pull/1282))
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:40]:
```suggestion
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:30]:
```suggestion
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:2]:
```suggestion

## Release v0.4.0 (2021/06/20)
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:10]: Wonder if we can come up with a more catchy highlight item :slightly_smiling_face: Probably list the datetime and vector tutorials? And we probably don't need to mention that there's 10 new wrapped functions since it's the next item down.

File: README.rst
Comment (weiji14) [inline README.rst:158]: > > > How do I know whether any of the current order reflects a case-by-case basis or whether I should update based on `git shortlog -sne`?
> > 
> > See previous discussion at [#1047 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1047#discussion_r593880470). Will leave it up to you to take a first stab at the order :) Just to clarify, that line refers to the Zenodo citation order. The release note authors order should go strictly by `git shortlog` (and remove the bots).
> 
> OK, I guess I wasn't paying good attention for that release :) Thanks for the note.

Just on the author list, this is the order from `git shortlog -sne` on the master branch (truncated to be for authors with >2 commits):

```
   279  Dongdong Tian <seisman.info@gmail.com>
   256  Leonardo Uieda <leouieda@gmail.com>
   141  Wei Ji <23487320+weiji14@users.noreply.github.com>
    83  Will Schlitzer <schlitzer90@gmail.com>
    51  Michael Grund <23025878+michaelgrund@users.noreply.github.com>
    36  Meghan Jones <meghanj@hawaii.edu>
    31  Yao Jiayuan <coreman.seism@gmail.com>
    27  dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
    15  Liam Toney <38269494+liamtoney@users.noreply.github.com>
     3  kmaterna <kathrynmaterna@gmail.com>
     2  Abhishek Anant <81291713+itsabhianant@users.noreply.github.com>
     2  Andrey Shmakov <akshmakov@gmail.com>
     2  Conor Bacon <conor.bacon@gmail.com>
     2  MichaeINeumann <81390813+MichaeINeumann@users.noreply.github.com>
     2  Noor Buchi <55197145+noorbuchi@users.noreply.github.com>
     2  Shivani chauhan <56837749+xdshivani@users.noreply.github.com>
     2  Tyler Newton <54011699+tjnewton@users.noreply.github.com>
     2  alperen-kilic <58446832+alperen-kilic@users.noreply.github.com>
```

Might need to shuffle some positions. And we'll need to add in Abhishek since he's on the AUTHORS.md (added in #1157).

**Review by weiji14 — COMMENTED** [review]
Another round of suggestions Meghan. Sorry if there's a lot of back and forth. This is why we want to cycle around the release manager task around the PyGMT team so that more people have experience :slightly_smiling_face: Plus we get to catch all the 'undocumented' procedures left out of https://github.com/GenericMappingTools/pygmt/blob/master/doc/maintenance.md#making-a-release.

**Comment by core-man** [comment]
> > How do I know whether any of the current order reflects a case-by-case basis or whether I should update based on `git shortlog -sne`?
> 
> See previous discussion at [#1047 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1047#discussion_r593880470). Will leave it up to you to take a first stab at the order :) Just to clarify, that line refers to the Zenodo citation order. The release note authors order should go strictly by `git shortlog` (and remove the bots).

Maybe we could add a sentence in [AUTHORSHIP.md#changelog-for-each-release](https://github.com/GenericMappingTools/pygmt/blob/master/AUTHORSHIP.md#changelog-for-each-release) to also clarify the author order in the release note. This will avoid confusion.

Also, simple examples to explain `The order can also be changed on a case-by-case basis` would also help, e.g., **Meghan** and **Liam**'s contributions, since new contributors might not know their great contributes to the PyGMT. This would help avoid confusion last time I made in v0.3.1 release changelog :smile:.

File: doc/changes.md
Comment (core-man) [reply doc/changes.md:10]: > Wonder if we can come up with a more catchy highlight item  Probably list the datetime and vector tutorials?

Totally agreed. Those two tutorials are great and very helpful. Perhaps we could highlight a few tutorial/gallery for each release.

> And we probably don't need to mention that there's 10 new wrapped functions since it's the next item down.

Agreed.

Comment (michaelgrund) [reply doc/changes.md:10]: The datetime and vector tutorials are definetly worth to mention them as highlights.

Comment (meghanrjones) [reply doc/changes.md]: Done in https://github.com/GenericMappingTools/pygmt/pull/1341/commits/3cdceb364d707c16bb4db23dc7adfd0abc164857

File: doc/changes.md
Comment (core-man) [inline doc/changes.md:56]: Sort the two terms alphabetically?
```suggestion
* NEP29: Set minimum required versions to NumPy 1.17+ and Python 3.7+ ([#1074](https://github.com/GenericMappingTools/pygmt/pull/1074))
* Raise a warning for the use of short-form parameters when long-forms are available ([#1316](https://github.com/GenericMappingTools/pygmt/pull/1316))
```

File: doc/changes.md
Comment (core-man) [inline doc/changes.md:45]: Sort those terms alphabetically?
```suggestion
* Add parameters to histogram ([#1249](https://github.com/GenericMappingTools/pygmt/pull/1249))
* Add alias 'aspatial' to methods blockmedian, info, plot, plot3d, surface ([#1090](https://github.com/GenericMappingTools/pygmt/pull/1090))
* Add alias 'registration' to methods blockmean, info, grdfilter, surface ([#1089](https://github.com/GenericMappingTools/pygmt/pull/1089))
* Add incols to COMMON_OPTIONS, blockmean, and blockmedian ([#1300](https://github.com/GenericMappingTools/pygmt/pull/1300))
* Improve Figure.show for displaying previews in Jupyter notebooks and external viewers ([#529](https://github.com/GenericMappingTools/pygmt/pull/529))
* Let Figure.savefig recommend .eps or .pdf when .ps extension is used ([#1307](https://github.com/GenericMappingTools/pygmt/pull/1307))
```

File: doc/changes.md
Comment (core-man) [inline doc/changes.md:63]: Sort the three terms alphabetically?
```suggestion
* Explicitly use netcdf4 engine in xarray.open_dataarray to read grd files ([#1264](https://github.com/GenericMappingTools/pygmt/pull/1264))
* Let Figure.savefig support filenames with spaces ([#1116](https://github.com/GenericMappingTools/pygmt/pull/1116))
* Let Figure.show(method='external') work well in Python scripts ([#1062](https://github.com/GenericMappingTools/pygmt/pull/1062))
```

File: doc/changes.md
Comment (core-man) [inline doc/changes.md:89]: Put revisions of **CONTRIBUTING.md** and **maintenance.md** together, respectively?
```suggestion
* Clarify interplay of spacing and per_column in info ([#1127](https://github.com/GenericMappingTools/pygmt/pull/1127))
* Remove the "full test" section from installation guide ([#1206](https://github.com/GenericMappingTools/pygmt/pull/1206))
* Add guidelines for managing issues to maintenance.md ([#1301](https://github.com/GenericMappingTools/pygmt/pull/1301))
* Clarify position of deprecate_parameter decorator to be above use_alias ([#1302](https://github.com/GenericMappingTools/pygmt/pull/1302))
* Add alias name convention to CONTRIBUTING.md ([#1256](https://github.com/GenericMappingTools/pygmt/pull/1256))
* Move contributing guide details to website and rename two sections ([#1335](https://github.com/GenericMappingTools/pygmt/pull/1335))
* Revise Pull Request review process in CONTRIBUTING.md ([#1119](https://github.com/GenericMappingTools/pygmt/pull/1119))
* Update the check_figures_equal testing section in CONTRIBUTING.md  ([#1108](https://github.com/GenericMappingTools/pygmt/pull/1108))
```

File: doc/changes.md
Comment (core-man) [inline doc/changes.md:70]: Move two new gallery examples to the top?
```suggestion
* Add histogram gallery example ([#1272](https://github.com/GenericMappingTools/pygmt/pull/1272))
* Add a gallery example showing individual basic geometric symbols ([#1211](https://github.com/GenericMappingTools/pygmt/pull/1211))
* Specify rectangle's width and height via style parameter in multi-parameter symbols example ([#1325](https://github.com/GenericMappingTools/pygmt/pull/1325))
* Update the inset gallery example ([#1287](https://github.com/GenericMappingTools/pygmt/pull/1287))
```

File: README.rst
Comment (meghanrjones) [reply README.rst:158]: Will update shortly.

Comment (meghanrjones) [reply README.rst:158]: Shuffled in https://github.com/GenericMappingTools/pygmt/pull/1341/commits/c2c210df1eed9f84249ffb10b56efc4dde670f72.

**Review by michaelgrund — APPROVED** [review]
From my side everything looks fine. All open points are considered. Great work!

**Review by seisman — APPROVED** [review]

**Review by core-man — APPROVED** [review]
I did not check the authors, but all the other looks great.

**Comment by meghanrjones** [comment]
Should I wait for Wei Ji to approve as well or is OK to move forward with the comments addressed and the three approvals?

**Comment by core-man** [comment]
> Should I wait for Wei Ji to approve as well or is OK to move forward with the comments addressed and the three approvals?

Perhaps it would better to wait for his approval too, since it does not matter to release v0.4 a few hours earlier or later.

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:100]: This entry feels like a new feature to me rather than a maintenance item. Could you move it to the 'feature' section please?

**Review by weiji14 — APPROVED** [review]
Just one suggestion, otherwise all good!

**Pull request merged**
